src/java/org/apache/cassandra/io/sstable/format/big/BigFormatV3.java

```diff
@@ -0,0 +1,309 @@
+    // we always incremented the major version.
+    static class BigVersion extends Version
+    {
+        public static final String current_version = "me";
+        public static final String earliest_supported_version = "jb";
+
+        // jb (2.0.1): switch from crc32 to adler32 for compression checksums
+        //             checksum the compressed data
+        // ka (2.1.0): new Statistics.db file format
+        //             index summaries can be downsampled and the sampling level is persisted
+        //             switch uncompressed checksums to adler32
+        //             tracks presense of legacy (local and remote) counter shards
+        // la (2.2.0): new file name format
+        // lb (2.2.7): commit log lower bound included
+        // ma (3.0.0): swap bf hash order
+        //             store rows natively
+        // mb (3.0.7, 3.7): commit log lower bound included
+        // mc (3.0.8, 3.9): commit log intervals included
+        // md (3.0.18, 3.11.4): corrected sstable min/max clustering
+        // me (3.0.25, 3.11.11): added hostId of the node from which the sstable originated
+        //
+        // NOTE: when adding a new version, please add that to LegacySSTableTest, too.
+
+        private final boolean isLatestVersion;
+        private final boolean hasSamplingLevel;
+        private final boolean newStatsFile;
+        private final ChecksumType compressedChecksumType;
+        private final ChecksumType uncompressedChecksumType;
+        private final boolean hasRepairedAt;
+        private final boolean tracksLegacyCounterShards;
+        private final boolean newFileName;
+        public final boolean storeRows;
+        public final int correspondingMessagingVersion; // Only use by storage that 'storeRows' so far
+        public final boolean hasBoundaries;
+        /**
+         * CASSANDRA-8413: 3.0 bloom filter representation changed (two longs just swapped)
+         * have no 'static' bits caused by using the same upper bits for both bloom filter and token distribution.
+         */
+        private final boolean hasOldBfHashOrder;
+        private final boolean hasCommitLogLowerBound;
+        private final boolean hasCommitLogIntervals;
+        private final boolean hasAccurateMinMax;
+        private final boolean hasOriginatingHostId;
+        private final boolean hasPendingRepair;
+        private final boolean hasMetadataChecksum;
+        private final boolean hasIsTransient;
+        /**
+         * CASSANDRA-7066: compaction ancerstors are no longer used and have been removed.
+         */
+        private final boolean hasCompactionAncestors;
+        /**
+         * CASSANDRA-9067: 4.0 bloom filter representation changed (two longs just swapped)
+         * have no 'static' bits caused by using the same upper bits for both bloom filter and token distribution.
+         */
+        private final boolean hasOldBfFormat;
+        BigVersion(String version)
```

Comment:
The logic here can be simplified to be just the flags from the `current_version`.  Otherwise this gives the impression it can write all these versions.  I would change the comments to make this clear